File: Application/CQRS/Narrators/Request-Response/NarratorDetailsResponse.cs
namespace Application.CQRS.Narrators
{
    public record NarratorDetailsResponse
  (
    int Id,
    int? Number,
    string? Name,
    string? Gender,
    string? Kunya,
    string? Nickname,
    string? Nasab,
    string? Description,
    string? Title,
    string? Activity,
    string? Madhhab,
    string? Rank,
    string? Layer,
    string? DeathYear,
    string? BirthYear,
    string? Age,
    string? Residence,
    string? DeathPlace,
    string? Relatives,
    string? Mawali,
    string? NarratedFor,
    string? Kamal,
    string? SirAlamAlNubala,
    string? SifatAlSafwa,
        List<NarratorStudentListResponse> NarratorStudents,
        List<NarratorTeacherListResponse> NarratorTeachers,
        List<NarratorsCriticismListResponse> NarratorsCriticisms

    );

    public record NarratorStudentListResponse
        (
            int Id,
         string Name,
         string? Kunya,
         string? Honorific,
         string? Lineage
    );
    public record NarratorTeacherListResponse
      (
      int Id,
     string Name,
   string? Kunya,
   string? Honorific,
   string? Lineage
  );
    public record NarratorsCriticismListResponse
      (int Id,
          string? CriticName,
     string? CriticStatement
  );
}

File: clientssr/src/app/Model/Narrators/narrator-details-response.ts
import { NarratorCriticismListResponse } from "./narrator-criticism-list-response";
import { NarratorStudentListResponse } from "./narrator-student-list-response";
import { NarratorTeacherListResponse } from "./narrator-teacher-list-response";



export interface NarratorDetailsResponse {
  id: number;
  number: number | null;
  name: string | null;
  gender: string | null;
  kunya: string | null;
  nickname: string | null;
  nasab: string | null;
  description: string | null;
  title: string | null;
  activity: string | null;
  madhhab: string | null;
  rank: string | null;
  layer: string | null;
  deathYear: string | null;
  birthYear: string | null;
  age: string | null;
  residence: string | null;
  deathPlace: string | null;
  relatives: string | null;
  mawali: string | null;
  narratedFor: string | null;
  kamal: string | null;
  sirAlamAlNubala: string | null;
  sifatAlSafwa: string | null;

  narratorStudents: NarratorStudentListResponse[];
  narratorTeachers: NarratorTeacherListResponse[];
  narratorsCriticisms: NarratorCriticismListResponse[];
}
